### ISSUE TITLE
Create components for Workspace Downgrade

### ISSUE THREAD

**Issue by carlosmiceli** [Expensify/App#51494]
Part of [Add Workspace Plan Downgrades to NewDot](https://github.com/Expensify/Expensify/issues/399936) 

[Doc section](https://docs.google.com/document/d/1GDPmbvV47VVxucr-tHYjH80VylcJptP0RfGMVy3zanY/edit?tab=t.0#bookmark=id.iwjdboyqgfbu)

Steps to complete:
- Update Workspace Profile Page
- Create Plan Type RHP
- Create Downgrade Confirmation RHP
- Create Downgrade Success RHP
- Create Downgrade Success RHP for admins of other Control workspaces

<details><summary>Issue Owner</summary>Current Issue Owner: @DylanDylann</details>

**Comment by DylanDylann** [comment]
@carlosmiceli Could we divide this issue into multiple PRs?

**Comment by DylanDylann** [comment]
@carlosmiceli Could you help to give a more explanation about the difference between these tasks?
- Create generic downgrade component
- Create WorkspaceDowngradePage

I notice that we need to create these tasks:
- Create Plan Type Page
- Create a generic success component
- Upgrade #admins messaging and workspace owner whisper

**Comment by carlosmiceli** [comment]
Sorry about the delay, have been sick the last couple of days 🙏 

I don't see why not have multiple PRs if you think that's best.

**Comment by carlosmiceli** [comment]
@DylanDylann isn't the success component the same as the confirmation component?

When you say plan type page, do you mean the workspace downgrade page? Btw, I think @jayeshmangwani already put that page together on the other PR, but since it's a PR about routes I think it shouldn't be included there. Can we move that into a the new PR that Dylan is going to work on so it's better organized according to the issues' scopes?

I'll create an issue about upgrades soon, but let me get back to you on that later this week 👍

**Comment by DylanDylann** [comment]
@carlosmiceli 

1. This is the plan type page

<img width="651" alt="Screenshot 2024-10-30 at 17 56 46" src="https://github.com/user-attachments/assets/b52e7701-9087-4dae-93be-f5c2f3dbb53c">

2. This is the workspace downgrade page

<img width="649" alt="Screenshot 2024-10-30 at 17 57 10" src="https://github.com/user-attachments/assets/ace61d3c-24d5-4f07-bc89-684e90121139">

3. This is the confirm/success page (confirm page and success page are same 😄 )

<img width="653" alt="Screenshot 2024-10-30 at 17 58 03" src="https://github.com/user-attachments/assets/57788cee-54be-458e-9bc2-a24158a14bc2">

4. Handle downgrade message in admin room

<img width="653" alt="Screenshot 2024-10-30 at 17 59 03" src="https://github.com/user-attachments/assets/39f2d574-0de9-44e7-8637-8510523e81cb">

**Comment by DylanDylann** [comment]
@carlosmiceli And in this issue, we will implement:
- The plan type page
- The downgrade page
- The confirm/success page

**Comment by dannymcclain** [comment]
Not sure if it's helpful or not, but the downgrade page and confirm/success page follow the same pattern/layout as our upgrade equivalent of those screens. So ideally we could reuse a lot of styles from those upgrade screens that are already built.

**Comment by carlosmiceli** [comment]
@dannymcclain Thanks Danny, good idea! 

@DylanDylann Yes, let's handle those three elements on this issue. We need to move the downgrade page from this [PR](https://github.com/Expensify/App/pull/51518#pullrequestreview-2404058838) into your PR(s) though, so it's better organized. cc @jayeshmangwani

**Comment by DylanDylann** [comment]
Sure 🚀

**Comment by carlosmiceli** [comment]
@DylanDylann thanks for checking me on the issue description, I just updated it according to this section and images in the doc, I think this will be clearer: https://docs.google.com/document/d/1GDPmbvV47VVxucr-tHYjH80VylcJptP0RfGMVy3zanY/edit?tab=t.0#bookmark=id.mnt3pa1wzv

It's 5 items now (do it in as many PRs as you'd like), I'll create another issue for the messaging portion.

**Comment by DylanDylann** [comment]
I am woking on PR, try to open PR on Friday

**Comment by carlosmiceli** [comment]
Great, thanks sir! I'll be ready to look at it today or during the weekend, let me know if you need anything :)

**Comment by DylanDylann** [comment]
Let's me know when the Downgrade API is done. Thanks

**Comment by jayeshmangwani** [comment]
@DylanDylann , have you created a page for the [plan type](https://github.com/Expensify/App/issues/51494#:~:text=This%20is%20the%20plan%20type%20page)? If so, please raise a PR so I can review it.

cc: @carlosmiceli

**Comment by DylanDylann** [comment]
This requires a BE update before we can implement the App PR

**Comment by jayeshmangwani** [comment]
You can raise a FE changes PR so I can use that page for the locked type. Please go ahead with the FE changes, and we can implement the BE later. What do you think, @carlosmiceli ?

**Comment by DylanDylann** [comment]
It required another PR to integrate API. Please create a PR for https://github.com/Expensify/App/issues/51768 and include the plan type on your PR.

When the API is done, I will raise PR for this issue and https://github.com/Expensify/App/issues/51764

**Comment by DylanDylann** [comment]
@jayeshmangwani Let's make it flexible to process the project asap

**Comment by jayeshmangwani** [comment]
Sounds good! I'll implement this in the PR today.

**Comment by jayeshmangwani** [comment]
>  Let's make it flexible to process the project asap

@DylanDylann sorry I didn't get this, I have blocked you in any way

**Comment by jayeshmangwani** [comment]
@DylanDylann You mentioned you were going to implement [this page](https://github.com/Expensify/App/issues/51494#issuecomment-2446601608), so I should wait for you, right?

**Comment by garrettmknight** [comment]
Small update, I just realized that the copy in the Figma doesn't match the copy we agreed on in the doc for the Plan type RHP. Can we update to:
> Choose a plan that's right for you. For a detailed list of features and pricing, check out our [plan types and pricing help page](https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/Plan-types-and-pricing).

Additionally, we slightly updated the URL to the pricing help page to https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/Plan-types-and-pricing

<img width="249" alt="Screenshot 2024-11-07 at 12 21 52 PM" src="https://github.com/user-attachments/assets/a4f39b44-ab30-4212-b045-a0b10f62fe57">

**Comment by jayeshmangwani** [comment]
> Can we update to:
Choose a plan that's right for you. For a detailed list of features and pricing, check out our [plan types and pricing help page](https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/Plan-types-and-pricing).

Sure, will update this copy and URL

**Comment by DylanDylann** [comment]
@carlosmiceli Have we completed API for Workspace downgrade?

**Comment by carlosmiceli** [comment]
Hey @DylanDylann sorry for being a bit delayed here, there are no updates yet because I'm tied down with a couple of critical issues. I'd definitely suggest to get the FE work done as advanced as possible before needing the API requests to work fully, so we can get a bit ahead. Do you need everything ready from the BE to make progress here?

**Comment by DylanDylann** [comment]
@carlosmiceli Any update on the BE changes

**Comment by carlosmiceli** [comment]
It's under review, hopefully merged soon! 🙏

**Comment by carlosmiceli** [comment]
Final BE PR has been merged :)

**Comment by DylanDylann** [comment]
@garrettmknight All done

**Comment by garrettmknight** [comment]
Sweet - I'm going to close this one then. Once @carlosmiceli gets back we'll work through all the PRs and pay out in bulk for the project.
